### PR TITLE
Support expression and selection plan

### PR DESCRIPTION
**Pull request by leiysky**
Initial implementation of expression. Also provide implementation of buiding selection plan.

### CODE REVIEW

File: src/planner/plan_builder.rs [hunk 1]
```diff
@@ -67,4 +71,24 @@ impl PlanBuilder {
...
+        let predicate = self.sql_expression_to_expr(selection)?;
+        Ok(PlanNode::Selection(SelectionPlan {
+            predicates: vec![predicate],
+        }))
+    }
+
+    fn sql_expression_to_expr(&self, expr: &ast::Expr) -> MySQLResult<Expr> {
+        match expr {
+            ast::Expr::BinaryOp { left, op, right } => match op {
+                ast::BinaryOperator::Eq => Ok(Expr::Equal {
```
Comment (andylokandy) [inline src/planner/plan_builder.rs:99]: This never works without columnref

Comment (leiysky) [reply src/planner/plan_builder.rs:99]: @andylokandy Yes, but there isn't `SELECT` statement yet. So even if `ColumnRef` has been supported, this never works either.

File: src/planner/plan_builder.rs [hunk 2]
```diff
@@ -1,13 +1,16 @@
 use crate::errors::{MySQLError, MySQLResult};
-use crate::planner::point_get_plan::QueryPlanBuilder;
+use crate::expression::Expr;
 use crate::planner::{CreateTablePlan, PlanNode, PointGetPlan, SelectPlan};
 use crate::session::SessionRef;
 use crate::table::schema::TableInfo;
+use sqlparser::ast;
```
Comment (zhongzc) [inline src/planner/plan_builder.rs:7]:
```suggestion

use sqlparser::ast;
```

File: src/planner/plan_builder.rs [hunk 3]
```diff
@@ -1,13 +1,16 @@
 use crate::errors::{MySQLError, MySQLResult};
-use crate::planner::point_get_plan::QueryPlanBuilder;
+use crate::expression::Expr;
```
Comment (zhongzc) [inline src/planner/plan_builder.rs:2]: There is `crate::planner::plan_expression`. Is it too hurry to convert AST to executable expression?

Comment (leiysky) [reply src/planner/plan_builder.rs:2]: > There is `crate::planner::plan_expression`. Is it too hurry to convert AST to executable expression?

Is there? I haven't seen that before.

By the way, this `Expr` is just a descriptor of expression, which is decoupled from execution. So I think it's fine to do this.

File: src/planner/plan_builder.rs (hunk 1 above)
Comment (Little-Wallace) [reply src/planner/plan_builder.rs:99]: What is ColumnRef?

**Pull request closed without merging**